api/app/Http/Controllers/AuthController.php

```diff
@@ -212,7 +217,52 @@ public function handleProviderCallback($provider, Socialite $socialite, UserRepo
     }
 
     /**
-     * Check so the provider exists.
+     * Provide a requester with user information for single sign on.
+     *
+     * @param  string  $requester
+     * @param  Request $request
+     *
+     * @return Response
+     */
+    public function singleSignOn($requester, Request $request)
+    {
+        if ($token = $request->cookie('JwtAuthToken')) {
+            $user = $this->jwtAuth->toUser($token);
+        } else {
+            $user = null;
```

Comment:
The response is for Vanilla, which wouldn't understand an Unauthorised exception. When the user is not logged in on the site it requests auth info from, Vanilla expects a response with empty information for the attributes.